### PR TITLE
Implement fanout support via SNS+SQS

### PR DESCRIPTION
**Pull request by kibertoad**
This allow to have publish/subscribe notifications for multiple consumers.

### CODE REVIEW

**Review by dariacm — APPROVED** [review]
Do I get it right that we are using SQS queues to subscribe to SNS topics? I don't have lots of experience with SNS/SQS, so maybe my comments are wrong. Feel free to dismiss in that case! 🙇🏻‍♀️ 

But overall looks good to me 💯

File: packages/sns/lib/sns/snsMessageDeserializer.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import { SNS_MESSAGE_BODY_SCHEMA } from '../types/MessageTypes'
+
+export const deserializeSNSMessage = <T extends object>(
+  message: SQSMessage,
+  type: ZodType<T>,
+  errorProcessor: ErrorResolver,
+): Either<MessageInvalidFormatError | MessageValidationError, T> => {
+  try {
+    const sqsMessage = SNS_MESSAGE_BODY_SCHEMA.parse(JSON.parse(message.Body))
```
[the remark this answers is not in the document]
Comment (kibertoad) [reply packages/sns/lib/sns/snsMessageDeserializer.ts:14]: Here it is actually correct, there is an SNS envelope packed inside an SQS message. I'll add comment to clarify that

File: packages/sns/lib/sns/AbstractSnsService.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export type SQSDependencies = QueueDependencies & {
+  snsClient: SNSClient
+}
+
+export type SNSConsumerDependencies = SQSDependencies & QueueConsumerDependencies
+
+export type SNSTopicAWSConfig = Omit<CreateTopicCommandInput, 'Name'>
+export type SQSQueueConfig = {
+  tags?: Record<string, string>
+  // ToDo if correct for sns
```
[the remark this answers is not in the document]
Comment (kibertoad) [reply packages/sns/lib/sns/AbstractSnsService.ts:20]: thank you for reminder, will revise

File: packages/sns/lib/sns/snsMessageDeserializer.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (kibertoad) [reply packages/sns/lib/sns/snsMessageDeserializer.ts]: No, you were right, it's actually an SNS message at that step, fixed

**Review by dariacm — APPROVED** [review]

**Pull request merged**